src/controllers/comments.controller.js

```diff
@@ -0,0 +1,152 @@
+import UserService from '../services/user.service';
+import models from '../models';
+const CommentsDb = models.Comment;
+const database = models.Article;
+/**
+ *
+ *
+ * @class Comments
+ */
+class Comments {
+  /**
+   *
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns {object} data
+   * @memberof Comments
+   */
+  static async createComment(req, res) {
+    const userId = req.auth.id;
+    const getUser = await UserService.getOneUser(userId);
+    const getArticle = await database.findOne({
+      where: { slug: req.params.slug }
+    });
+    if (!getUser) return res.status(404).send({ message: `User with id ${req.auth.id} not found` });
+    if (!getArticle) return res.status(404).send({ message: `Article with slug ${req.params.slug}  not found` });
+    try {
+      const comment = {
+        articleSlug: getArticle.slug,
+        userId: req.auth.id,
+        body: req.body.body,
+        parentCommentId: req.body.parentCommentId,
+      };
+      const createdComment = await commentsService.addComment(comment);
+      return res.status(201).send({
+        status: 201,
+        message: 'Comment successfully created',
+        comment: createdComment
+      });
+    } catch (error) {
+      return res.send({
+        message: error.message
+      });
+    }
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns {object} data
+   * @memberof Comments
+   */
+  static async deleteComment(req, res) {
+    const userId = req.auth.id;
+    const getUser = await CommentsDb.findOne({ where: { userId } });
+    const commentAuthor = getUser && getUser.get().userId;
+    const { id } = req.params;
+
+    if (!Number(id)) {
+      return res.status(400).send({
+        status: 400,
+        message: 'Please provide numeric value'
+      });
+    }
+    if (!(req.auth.id === commentAuthor)) {
+      return res.status(403).send({
+        status: 403,
+        message: 'comment is not yours'
+      });
+    }
+    try {
+      const CommentTODelete = await commentsService.deleteComment(id);
+      if (CommentTODelete) {
+        return res.status(200).send({
+          status: 200,
+          message: `Comment with id ${id} is successfully deleted`
+        });
+      }
+
+      return res.status(404).send({
+        status: 404,
+        message: `Comment with id ${id} is not found`
+      });
+    } catch (error) {
+      return res.send({
+        message: error.message
+      });
+    }
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns {object} data
+   * @memberof Comments
+   */
+  static async getComments(req, res) {
+    const comments = await CommentsDb.findAll();
+    if (!comments) {
+      return res.status(200).send({
+        message: 'No comments found'
+      });
+    }
+    return res.status(200).send({
+      status: 200,
+      message: 'All comments successfully retrieved',
+      comments
```

Comment:
@nshutijonathan For the response format we agreed to be in this format:
{
  status: 200,
  message: 'All comments successfully retrieved',
  data: comments
}
However @bdushimi suggested we remove the status attribute. I would like to know which  format our response for success response we finally decided to follow

Reply:
Yes I recall this, but We have concluded to come up the techniques of utils for this upcoming sprint and We shall raise the chore of refactoring all codes and use utils as well, I think you agree with me

Reply:
Oh okay. Cool